scripts/functional-tests.sh

```diff
@@ -8,6 +14,11 @@ export SCRIPTDIR=$(dirname "$0")
 # build a prompt string that includes the time, source file, line number, and function name
 export PS4='+$(date +"%Y-%m-%d %T") ${BASH_VERSION}:${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
+# Support overriding the test directory for local testing otherwise creates a temporary directory
+TEST_DIR=${TEST_DIR:-$(mktemp -d)}
```

Comment:
There's another issue with the bare call to `mktemp` here - it will use `/tmp` (unless `$TMPDIR` is set to point to elsewhere.) And this directory is not meant for large files (e.g. on my vagrant box with Fedora 40, I get only 2GB available in `/tmp` - which makes the functional test runs fail with `No space left on device` because the model is 4GB+.

Quoting from [here](https://systemd.io/TEMPORARY_DIRECTORIES/):

```
/tmp/ should be used for smaller, size-bounded files only; /var/tmp/ should be used for everything else.
```

Reply:
Added in a separate commit, thanks!